contrib/win32/win32compat/misc.c

```diff
@@ -283,33 +283,49 @@ char*
  w32_fgets(char *str, int n, FILE *stream) {
 	HANDLE h = (HANDLE)_get_osfhandle(_fileno(stream));
 	wchar_t* str_w = NULL;
-	char *ret = NULL, *str_tmp = NULL;
+	char *ret = NULL, *str_tmp = NULL, *cp = NULL;
```

Comment:
I mean, ret variable is not required.. we can still use str variable itself..at the start of the function initialize str to NULL and everywhere return str..
Its a minor comment, fix as you like.